specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts

```diff
@@ -183,7 +193,9 @@ export class BusinessRuleManager<SCHEMA extends AnySchema> {
       )
     );
 
-    const results = rulesToCheck.map(async ({ rule }) => this.uniqueIn(rule));
+    const results: RA<Promise<BusinessRuleResult<SCHEMA>>> = rulesToCheck.map(
+      async ({ rule }) => this.uniqueIn(rule)
+    );
     void Promise.all(results).then((results) =>
```

Comment:
huh, it's weird that a function returns an array of promises rather than a single promise from Promise.all
why not do this instead?

```ts
const results = await Promise.all(results);

results.flatMap ... do something with results

return results;
```

also, since your checkUnique doesn't await the promise before returning the result, it may be possible for it to be called again before the previous call completed, thus creating a race condition